docs/source/en/using-diffusers/loading.mdx

```diff
@@ -178,105 +198,177 @@ stable_diffusion_img2img = StableDiffusionImg2ImgPipeline(**components)
+
+Pipeline checkpoints that are identical in checkpoint layout, but have been trained on different datasets, trained with vastly diffenent training setups and thus correspond to different official releases (such as [Stable Diffusion v1-4](https://huggingface.co/CompVis/stable-diffusion-v1-4) and [Stable Diffusion v1-5](https://huggingface.co/runwayml/stable-diffusion-v1-5)) should probably also not be stored as variations of each other, but rather as individual model repositories.
+
+#### So what are checkpoint variants then? 
+
+Checkpoint variants usually consists of the checkpoint stored in "*low-precision, low-storage*" dtype so that less bandwith is required to download them or of *non-exponential-averaged* weights that shall be used when continuing fine-tuning the checkpoint.
+Both such use case have clear advantages in certain settings, have the same serialization format, and correspond to an adaption of the "main" checkpoint which does not warrant a new model repository. 
+A checkpoint stored in [torch's half-precision / float16 format](https://pytorch.org/blog/accelerating-training-on-nvidia-gpus-with-pytorch-automatic-mixed-precision/) requires only half the bandwith and storage when downloading the checkpoint,
+**but** cannot be used when continuing training the checkpoint or when running the checkpoint on CPU.
+Similarly the *non-exponential-averaged* version of the checkpoint should be used when continuing fine-tuning of the model checkpoint, **but** should not be used when using the checkpoint for inference. 
+
+#### How to save and load variants
+
+Saving a diffusion pipeline as a variant can be done by providing [`DiffusionPipeline.save_pretrained`] with the `variant` argument. 
+The `variant` extends the weight name by the provided variation, by changing the default weight name from `diffusion_pytorch_model.bin` to `diffusion_pytorch_model.{variant}.bin` or from `diffusion_pytorch_model.safetensors` to `diffusion_pytorch_model.{variant}.safetensors`. By doing so, one creates a variant of the pipeline checkpoint that can be loaded **instead** of the "main" pipeline checkpoint.
+
+Let's have a look at how we could create a float16 variant of a pipeline. First, we load 
+the "main" variant of a checkpoint into mixed precision format, using `dtype=torch.float16`.
+
+```py
+from diffusers import DiffusionPipeline
+import torch
+
+pipe = DiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", dtype=torch.float16)
+```
+
+Now all model components of the pipeline are stored in half-precision dtype. We can now save the 
+pipeline under a `"fp16"` variant as follows:
+
+```py
+pipe.save_pretrained("./stable-diffusion-v1-5", variant="fp16")
+```
+
```

Comment:
If you check the size of the output directory you'll notice that the weights take half the space as their counterparts in the Hub, as they were converted from `float32` to `float16`. You'll also see that the variant was added to the filenames so both versions can coexist in the same repo.